src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
+                    topTruncatedCounts + bottomTruncatedCounts, totalCounts, truncatedPercentage, bottomPercentileThreshold, topPercentileThreshold));
+        }
+    }
+
+    /**
+     * Creates a new read-count collection that is a copy of the input but dropping columns with extreme median coverage.
+     *
+     * @param readCounts the input read-counts.
+     * @param extremeColumnMedianCountPercentileThreshold bottom percentile to use as an exclusion threshold.
+     * @return never {@code null}. It might be a reference to the input read-counts if
+     * there are no columns to be dropped.
+     */
+    public static ReadCountCollection removeColumnsWithExtremeMedianCounts(final ReadCountCollection readCounts, final double extremeColumnMedianCountPercentileThreshold, final Logger logger) {
+        final List<String> columnNames = readCounts.columnNames();
+        final RealMatrix counts = readCounts.counts();
+        final Median medianCalculator = new Median();
+        final double[] columnMedians = IntStream.range(0, counts.getColumnDimension())
+                .mapToDouble(col -> medianCalculator.evaluate(counts.getColumn(col))).toArray();
+
+        // Calculate thresholds:
+        final double bottomExtremeThreshold = new Percentile(extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+        final double topExtremeThreshold = new Percentile(100 - extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new HashSet<>(readCounts.columnNames().size());
+        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
+        for (int i = 0; i < columnMedians.length; i++) {
+            if (columnMedians[i] >= bottomExtremeThreshold && columnMedians[i] <= topExtremeThreshold) {
+                columnsToKeep.add(columnNames.get(i));
+            } else {
+                columnsToDrop.put(columnNames.get(i), columnMedians[i]);
+            }
+        }
 
+        // log and drop columns if it applies
+        if (columnsToKeep.isEmpty()) {
+            throw new UserException.BadInput("No column count left after applying the extreme counts outlier filter");
+        } else if (columnsToKeep.size() == columnNames.size()) {
+            logger.info(String.format("No column dropped due to extreme counts outside [%.10f, %.10f]", bottomExtremeThreshold, topExtremeThreshold));
+            return readCounts;
+        } else {
+            logger.info(String.format("Some columns dropped, %d of %d, as they are classified as having extreme median counts across targets outside [%.10f, %.10f]: e.g. %s",
+                    columnNames.size() - columnsToKeep.size(), columnNames.size(), bottomExtremeThreshold, topExtremeThreshold,
```

Comment:
Why not use `columnsToDrop.size()` instead of `columnNames.size() - columnsToKeep.size()` ?